decision_service/dsclientlib/ds_config_collection.cc

```diff
@@ -0,0 +1,22 @@
+#include "ds_config_collection.h"
+
+namespace decision_service { namespace utility {
+
+  void config_collection::set(const char* name, const char* value) {
+    _map[name] = value;
```

Comment:
This seems semantically dangerous because the map has keys and values that can be mutated externally.

Reply:
there's a copy happening here due to map<string, string>

---
In reply to: [186774583](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r186774583) [](ancestors = 186774583)

Reply:
Right, thanks.